lib/llhttp/parser.js

```diff
@@ -0,0 +1,240 @@
+'use strict'
+
+/* global WebAssembly */
+
+const { resolve } = require('path')
+const { readFileSync } = require('fs')
+const constants = require('./constants')
+const { kMaxHeadersSize } = require('../core/symbols')
+const WASM_BUILD = resolve(__dirname, './llhttp.wasm')
+const bin = readFileSync(WASM_BUILD)
+const mod = new WebAssembly.Module(bin)
+
+const kOnMessageBegin = 0
+const kOnHeaders = 1
+const kOnHeadersComplete = 2
+const kOnBody = 3
+const kOnMessageComplete = 4
+const kOnExecute = 5
+
+const kPtr = Symbol('kPrt')
+const kStatusMessage = Symbol('kStatusMessage')
+const kHeadersFields = Symbol('kHeadersFields')
+const kHeadersValues = Symbol('kHeadersValues')
+const kBufferSize = Symbol('kBufferSize')
+const kHeadersTimeout = Symbol('kHeadersTimeout')
+const kBufferPtr = Symbol('kBufferPtr')
+const kMallocBuffer = Symbol('kMallocBuffer')
+const kCurrentBuffer = Symbol('kCurrentBuffer')
+const kGetHeaders = Symbol('kGetHeaders')
+const kMakeError = Symbol('kMakeError')
+
+/**
+ * Map of HTTPParser instances
+ */
+const instMap = new Map()
+
+const cstr = (ptr, len) =>
+  Buffer.from(inst.exports.memory.buffer, ptr, len).toString()
+
+/* eslint-disable camelcase */
+const wasm_on_message_begin = p => {
+  const i = instMap.get(p)
+  this[kStatusMessage] = null
+  return i[kOnMessageBegin]()
```

Comment:
```suggestion
  this[kStatusMessage] = null
  return instMap.get(p)[kOnMessageBegin]()
```

Reply:
On this one I made a mistake, `this` should be `i`.